library/Icingadb/Data/JsonResultSetUtils.php

```diff
@@ -70,7 +70,7 @@ public static function stream(Query $query): void
         if ($query->hasLimit()) {
             // Custom limits should still apply
             $query->peekAhead(false);
-            $offset = $query->getOffset();
+            $offset = $query->getOffset() ?: 0;
```

Comment:
Why not `??`?

Reply:
In this case, I see no advantage/disadvantage, as the method always returns a value. `(null|int`).

Reply:
I do. `0 ?? 0` only checks the type whereas `0 ?: 0` also the value. And since the method only returns a nullable int and a zero is fine, the latter is less efficient. But it's not about efficiency, but about clarity.

The next time you write something like this, and your fallback value doesn't match the default, this will bite you: `$value ?: 100`

In short: If a method may return null and you are fine with anything else, use `??`.

Reply:
I see, thanks for the explanation.